src/utils/GpioInitializer.hxx

```diff
@@ -61,22 +61,22 @@ template <typename Head, typename... Tail>
 struct GpioInitHelper<std::tuple<Head, Tail...>>
 {
     /// Middle step of the recursion for an individual pin.
-    static void hw_init()
+    static void __attribute__((always_inline)) hw_init()
     {
         Head::hw_init();
-        GpioInitHelper<std::tuple<Tail...>>::hw_init();
+        return GpioInitHelper<std::tuple<Tail...>>::hw_init();
```

Comment:
should we add `#include <tuple>` in this file to fix the missing declaration?

Reply:
done, good catch